### DISCUSSION TITLE
Audio Transcription

### DISCUSSION THREAD

**Discussion by thiagocaparelli** [sepinf-inc/IPED#648] [Q&A]
How do we implement the Audio Transcription functionality. I've a plenty of audios (ogg format) in a folder, but I don't know how to integrate Azure or Google Cloud with them. 

Thanks.

**Comment by lfcnassif** [comment]
You should follow Microsoft/Google documentation to setup those services, then configure the credentials as explained in  https://github.com/sepinf-inc/IPED/blob/master/iped-app/resources/config/conf/AudioTranscriptConfig.txt

Maybe @carlostada could help you, as he has set up Microsoft speech-to-text service very recently.

**Reply by lfcnassif** [reply]
PS: In the past Google service was reaching quota limits much before than it should (at least in Brazil), I still need to test the non beta libraries as suggested by them: https://github.com/googleapis/java-speech/issues/171

So I suggest trying Microsoft Azure for now.

**Reply by thiagocaparelli** [reply]
First of all, thank you.

I created a valuation account on Azure and got the subscription key.

Applied it like you suggested via command line, but I've got some errors.

(...)
2021-07-07 12:29:00	[INFO]	[indexer.process.Worker]			Starting AudioTranscriptTask
Could not copy external Speech SDK libraries because of the following error: null
Could not load an optional Speech SDK library because of the following error: C:\Users\Thiago\AppData\Local\Temp\indexador-temp1625671713888\speech-sdk-native-9028817658344062601\Microsoft.CognitiveServices.Speech.extension.codec.dll: Can't find dependent libraries
(...)

2021-07-07 12:29:41	[ERROR]	[task.transcript.MicrosoftTranscriptTask]			Transcription of ArquivosConvertidos\audio_102@28-03-2021_18-10-15.ogg.wav failed errorCode=ConnectionFailure details=Connection failed (no connection to the remote host). Internal error: 1. Error details: 954. Please check network connection, firewall setting, and the region name used to create speech factory. SessionId: fffd40e508a5486abe53234b17a18c63

(...)

Do you know what could be these errors?

**Reply by lfcnassif** [reply]
If you used the evaluation key that doesn't need a credit card, I think you should change serviceRegion to westus (or eastus, I don't remember) in config file, check your key details at your Azure account. But that allows just one transcription at a time, so you should also decrease maxConcurrentRequests to 1 in config, or you will get a lot of quota errors.
If you use a credit card, Microsoft used to give $200 in the first month which were enough for 200h of audios. They say they won't charge you automatically if your free credits are spent completely.

**Reply by thiagocaparelli** [reply]
Actually I put my credit card and got the 200$ free for testing. Also I changed the file to region "brazilsouth" or something like that. 
I copied that region like it appears on the azure site.

**Reply by lfcnassif** [reply]
So possibly this is a firewall problem...

**Reply by lfcnassif** [reply]
Ah... MS SDK doesn't work on Windows 7

**Reply by thiagocaparelli** [reply]
I'm running on Windows 10. 
I can't do it right now but I will stop de antivirus and try it again. 
I'm wondering now if it is needed some optional .jar

Thank you so much.

**Reply by lfcnassif** [reply]
Maybe this is related to missing dll's (first error), perhaps removed by the antivirus, or missing in your machine. If disabling antivirus and firewall don't help and If you can share the full log, I can take a look when some time is available.

**Reply by thiagocaparelli** [reply]
I realized that everytime I run the command suddenly my internet drop for DNS failure. 
But I still can ping outside. Just the DNS service stop.

I stopped the windows defender and the malwarebytes softwares. 

It's very strange.

[IPED-2021-07-08-13-18-32.log](https://github.com/sepinf-inc/IPED/files/6785950/IPED-2021-07-08-13-18-32.log)

**Reply by lfcnassif** [reply]
I have just checked Microsoft.CognitiveServices.Speech.extension.codec.dll embedded in the library jar and it depends on Visual Studio **msvcp140.dll** and **vcruntime140.dll**. I'm attaching those dll's below, try to put them into C:\Windows\system32 and see if that works.

[vs_dlls.zip](https://github.com/sepinf-inc/IPED/files/6786026/vs_dlls.zip)

**Reply by thiagocaparelli** [reply]
Thank you so much.

I already had these DLLs because of the installed packeges MS SDK and Visual Studio 2015-2019.

I don't know why but I closed every browser (mozilla and chrome) and It worked fine. The internet not droped anymore...

I ran wireshark to capture every DNS call to azure/microsoft and put them on hosts file, so I had to close all software that could generate garbage (including the browsers). Then after that it run without problem.
